contracts/protocore/Protocore.sol

```diff
@@ -17,27 +17,20 @@ pragma solidity >=0.5.0 <0.6.0;
 import "openzeppelin-solidity/contracts/math/SafeMath.sol";
 
 import "../consensus/CoconsensusModule.sol";
+import "../validator/ValidatorSet.sol";
 import "../version/MosaicVersion.sol";
 
 /**
  * @title Protocore abstract contract acting as a base contract for
  *        OriginProtocore and SelfProtocore contracts.
  */
-contract Protocore is MosaicVersion, CoconsensusModule {
+contract Protocore is MosaicVersion, ValidatorSet, CoconsensusModule {
```

Comment:
`SelfProtocore` should inherit `ValidatorSet`?

`OriginProtocore` and `SelfProtocore` both inherit `Protocore`. As per the specifications `OriginProtocore` should query the validator set related information from `SelfProtocore`.

`Protocore` can be abstract and have empty function `inForwardValidatorSet`.
`OriginProtocore` and `SelfProtocore` both can implement its logic by overriding it.

Reply:
Done.